src/main/java/ru/practicum/shareit/booking/Service/BookingServiceImpl.java

```diff
@@ -0,0 +1,298 @@
+            throw new InvalidResourceException(
+                    String.format("Пользователь c id = '%d' не владелец вещи и не создатель бронирования", userId));
+        }
+
+        if ((userId != bookingItemOwnerId)) {
+            throw new ResourceNotFoundException(
+                    String.format("Пользователь c id = '%d' не является владельцем вещи", userId));
+        }
+
+        if (approved) {
+            if (bookingToUpdate.getStatus().equals("APPROVED")) {
+                throw new InvalidResourceException(
+                        String.format("Бронирование  c id = '%d' уже одобрено владельцем вещи", bookingId));
+            }
+            bookingToUpdate.setStatus(Status.APPROVED.name());
+        } else {
+            bookingToUpdate.setStatus(Status.REJECTED.name());
+        }
+
+        bookingRepository.save(bookingToUpdate);
+
+        ItemOutputDtoForBooking returnItemDto
+                = ItemMapper.INSTANCE.itemToItemOutputDtoForBooking(bookingToUpdate.getItem());
+
+        UserOutputDtoForBooking returnUser
+                = UserMapper.INSTANCE.userToUserOutputDtoForBooking(bookingToUpdate.getBooker());
+
+        return BookingMapper.INSTANCE.bookingToBookingOutputDto(bookingToUpdate, returnItemDto, returnUser);
+    }
+
+    @Override
+    public BookingOutputDto getBookingsById(Long bookingId, Long userId) {
+        log.info("Service: Получение списка бронирований пользователя");
+
+        checkUserExistence(userId);
+
+        if (bookingId == null) {
+            throw new ResourceNotFoundException(
+                    "Id бронирования не задано");
+        }
+
+        Booking requestedBooking = bookingRepository.findById(bookingId).orElseThrow(
+                () -> new ResourceNotFoundException(
+                        String.format("Бронирование c id = '%d' не найдено", bookingId))
+        );
+
+
+        long bookerId = requestedBooking.getBooker().getId();
+        long bookingItemOwnerId = requestedBooking.getItem().getOwnerId();
+
+        if (((userId != bookerId)) && (userId != bookingItemOwnerId)) {
+            throw new ResourceNotFoundException(
+                    String.format("Пользователь c id = '%d' не имет прав для просмотра бронирования с id = '%d'", userId, bookingId));
+        }
+
+        ItemOutputDtoForBooking returnItemDto
+                = ItemMapper.INSTANCE.itemToItemOutputDtoForBooking(requestedBooking.getItem());
+
+        UserOutputDtoForBooking returnUser
+                = UserMapper.INSTANCE.userToUserOutputDtoForBooking(requestedBooking.getBooker());
+
+
+        return BookingMapper.INSTANCE.bookingToBookingOutputDto(requestedBooking, returnItemDto, returnUser);
+    }
+
+    @Override
+    public List<BookingOutputDto> getBookingsByBookerIdAndBookingState(String state, Long bookerId) {
+
+        checkUserExistence(bookerId);
+        State requestState;
+        requestState = getState(state);
```

Comment:
Нет смысла разбивать на две строчки
```suggestion
        State requestState = getState(state);
```

Reply:
Испарвил ошибочное разделение объявления и инициализации переменной